app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java

```diff
@@ -66,48 +79,127 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         SupportMapFragment mapFragment = (SupportMapFragment) this.getChildFragmentManager()
                 .findFragmentById(R.id.map);
         mapFragment.getMapAsync(this);
+        mQueue = HttpClient.getInstance(getActivity().getApplicationContext()).getRequestQueue();
+        mGson = new Gson();
+        mRelativeLayout = (RelativeLayout) view.findViewById(R.id.driver_home_layout);
+        mCurrentSpot = null;
 
         mQueue = HttpClient.getInstance(getActivity().getApplicationContext()).getRequestQueue();
         mFindParkingButton = (Button) view.findViewById(R.id.findparkingbutton);
         mFindParkingButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                // Reserve a spot for the user.
-                requestReservation(new HttpClient.VolleyCallback<String>() {
+                // If the button is in 'find spot' mode, reserve a spot
+                if (getString(R.string.find_spot_button_text).equals(mFindParkingButton.getText())) {
+                    // Reserve a spot for the user.
+                    reserveSpot();
+                } else {
+                    // Otherwise, navigate to the spot
+                    // TODO(kenzshelley) actually send an intent to navigate to the currently reserved spot.
+                    if (mCurrentSpot == null) {
+                        Log.d(CLASS, "No spot has been reserved!");
+                    }
+                }
+            }
+        });
+
+        return view;
+    }
+
+    private void reserveSpot() {
+        requestReservation(new HttpClient.VolleyCallback<String>() {
+            @Override
+            public void onSuccess(String response) {
```

Comment:
Let's leave it for now, but I think the case could be made for making separate functions that you would call in each onSuccess(). Sure you'd have to do some scrolling, but they'd all be in the same class right?